app/src/test/java/org/openobservatory/ooniprobe/domain/GenerateAutoRunServiceSuiteTest.java

```diff
@@ -0,0 +1,124 @@
+package org.openobservatory.ooniprobe.domain;
+
+import androidx.test.filters.SmallTest;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.openobservatory.engine.OONICheckInConfig;
+import org.openobservatory.engine.OONICheckInResults;
+import org.openobservatory.engine.OONICheckInResults.OONICheckInInfoWebConnectivity;
+import org.openobservatory.engine.OONISession;
+import org.openobservatory.engine.OONIURLInfo;
+import org.openobservatory.ooniprobe.RobolectricAbstractTest;
+import org.openobservatory.ooniprobe.common.PreferenceManager;
+import org.openobservatory.ooniprobe.engine.TestEngineInterface;
+import org.openobservatory.ooniprobe.test.EngineInterface;
+import org.openobservatory.ooniprobe.test.EngineProvider;
+import org.openobservatory.ooniprobe.test.suite.AbstractSuite;
+import org.openobservatory.ooniprobe.test.test.AbstractTest;
+
+import java.util.ArrayList;
+
+import io.bloco.faker.Faker;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@SmallTest
+public class GenerateAutoRunServiceSuiteTest extends RobolectricAbstractTest {
+
+    // Mocks
+    OONICheckInInfoWebConnectivity webConnectivityMock = mock(OONICheckInInfoWebConnectivity.class);
+    PreferenceManager preferenceManagerMock = mock(PreferenceManager.class);
+    OONICheckInConfig ooniCheckConfigMock = mock(OONICheckInConfig.class);
+    OONICheckInResults ooniResultsMock = mock(OONICheckInResults.class);
+    OONISession ooniSessionMock = mock(OONISession.class);
+
+    // Engine && UseCase
+    EngineInterface mockedEngine = new TestEngineInterface(ooniSessionMock);
+    GenerateAutoRunServiceSuite generateSuite;
+
+    @Override
+    public void setUp() {
+        super.setUp();
+        generateSuite = new GenerateAutoRunServiceSuite(a, preferenceManagerMock);
+        EngineProvider.engineInterface = mockedEngine;
+
+        when(preferenceManagerMock.testWifiOnly()).thenReturn(true);
+        when(preferenceManagerMock.testChargingOnly()).thenReturn(true);
+    }
+
+    @Test
+    public void shouldNotStartTest() {
+        // Act
+        AbstractSuite suite = generateSuite.generate(ooniCheckConfigMock, false, false);
+
+        // Assert
+        Assert.assertNull(suite);
+    }
+
+    @Test
+    public void generateSuite() throws Exception {
+        // Arrange
+        ArrayList<OONIURLInfo> suiteUrls = getTestUrls();
+
+        when(preferenceManagerMock.getEnabledCategoryArr()).thenReturn(new ArrayList<String>() {
+            {
+                add("ALDR");
+                add("REL");
+                add("PORN");
+                add("PROV");
+                add("POLR");
+                add("HUMR");
+                add("ENV");
+            }
+        });
```

Comment:
`Arrays.asList(...)` makes this slightly simpler.

Reply:
would require casting to ArrayList...